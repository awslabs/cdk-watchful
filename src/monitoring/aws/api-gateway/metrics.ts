import { Metric, Statistic } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';
import { WatchedOperation } from '../../../api-gateway';

const enum Metrics {
  FourHundredError = '4XXError',
  FiveHundredError = '5XXError',
  CacheHitCount = 'CacheHitCount',
  CacheMissCount = 'CacheMissCount',
  Count = 'Count',
  IntegrationLatency = 'IntegrationLatency',
  Latency = 'Latency',
}

const Namespace = 'AWS/ApiGateway';

export class ApiGatewayMetricFactory {
  metricErrors(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      errors4XX: this.metric(Metrics.FourHundredError, apiName, stage, op).with({ label: '4XX Errors', statistic: Statistic.SUM }),
      errors5XX: this.metric(Metrics.FiveHundredError, apiName, stage, op).with({ label: '5XX Errors', statistic: Statistic.SUM }),
    };
  }

  metricCache(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      hits: this.metric(Metrics.CacheHitCount, apiName, stage, op).with({ label: 'Cache Hit', statistic: Statistic.SUM }),
      misses: this.metric(Metrics.CacheMissCount, apiName, stage, op).with({ label: 'Cache Miss', statistic: Statistic.SUM }),
    };
  }

  metricCalls(apiName: string, stage: string, op?: WatchedOperation) {
    return this.metric(Metrics.Count, apiName, stage, op).with({ label: 'Calls', statistic: Statistic.SUM });
  }

  metricIntegrationLatency(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      min: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'min', statistic: Statistic.MINIMUM }),
      avg: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'avg', statistic: Statistic.AVERAGE }),
      p90: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'p90', statistic: 'p90' }),
      p95: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'p95', statistic: 'p95' }),
      p99: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'p99', statistic: 'p99' }),
      max: this.metric(Metrics.IntegrationLatency, apiName, stage, op).with({ label: 'max', statistic: Statistic.MAXIMUM }),
    };
  }

  metricLatency(apiName: string, stage: string, op?: WatchedOperation) {
    return {
      min: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'min', statistic: Statistic.MINIMUM }),
      avg: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'avg', statistic: Statistic.AVERAGE }),
      p90: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'p90', statistic: 'p90' }),
      p95: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'p95', statistic: 'p95' }),
      p99: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'p99', statistic: 'p99' }),
      max: this.metric(Metrics.Latency, apiName, stage, op).with({ label: 'max', statistic: Statistic.MAXIMUM }),
    };
  }

  protected metric(metricName: Metrics, apiName: string, stage: string, op?: WatchedOperation) {
    return new Metric({
      metricName,
      namespace: Namespace,
      period: Duration.minutes(1),
      dimensions: {
        ApiName: apiName,
        Stage: stage,
        ...op && {
          Method: op.httpMethod,
          Resource: op.resourcePath,
        },
      },
    });
  }
}

