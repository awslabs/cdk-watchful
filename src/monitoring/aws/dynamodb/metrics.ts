import { Metric } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';

export class DynamoDbMetricFactory {
  static readonly Namespace = 'AWS/DynamoDB';

  metricConsumedCapacityUnits(tableName: string) {
    function metric(metricName: string, label: string) {
      return new Metric({
        metricName,
        label,
        namespace: DynamoDbMetricFactory.Namespace,
        period: Duration.minutes(1),
        statistic: 'Sum',
        dimensions: {
          TableName: tableName,
        },
      });
    }

    return {
      read: metric('ConsumedReadCapacityUnits', 'Consumed (Read)'),
      write: metric('ConsumedWriteCapacityUnits', 'Consumed (Write)'),
    };
  }
}
