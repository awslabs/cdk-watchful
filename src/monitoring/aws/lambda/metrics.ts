import { Metric } from '@aws-cdk/aws-cloudwatch';
import * as cdk from '@aws-cdk/core';

export class LambdaMetricFactory {
  static readonly Namespace = 'AWS/Lambda';

  metricInvocations(functionName: string) {
    return new Metric({
      metricName: 'Invocations',
      namespace: LambdaMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        FunctionName: functionName,
      },
    });
  }

  metricDuration(functionName: string) {
    function metric(statistic: string) {
      return new Metric({
        metricName: 'Duration',
        label: statistic,
        namespace: LambdaMetricFactory.Namespace,
        period: cdk.Duration.minutes(5),
        statistic,
        dimensions: {
          FunctionName: functionName,
        },
      });
    }

    return {
      min: metric('Minimum'),
      avg: metric('Average'),
      p50: metric('p50'),
      p90: metric('p90'),
      p99: metric('p99'),
      max: metric('Maximum'),
    };
  }

  metricErrors(functionName: string) {
    return new Metric({
      metricName: 'Errors',
      namespace: LambdaMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        FunctionName: functionName,
      },
    });
  }

  metricThrottles(functionName: string) {
    return new Metric({
      metricName: 'Throttles',
      namespace: LambdaMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        FunctionName: functionName,
      },
    });
  }
}
