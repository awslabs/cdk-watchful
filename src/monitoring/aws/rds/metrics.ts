import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
import * as cdk from '@aws-cdk/core';

/**
 * Metrics for RDS Aurora.
 */
export class RdsAuroraMetricFactory {
  static readonly Namespace = 'AWS/RDS';

  metricDmlThroughput(clusterIdentifier: string) {
    const dbInsertThroughputMetric = new cloudwatch.Metric({
      metricName: 'InsertThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    const dbUpdateThroughputMetric = new cloudwatch.Metric({
      metricName: 'UpdateThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    const dbSelectThroughputMetric = new cloudwatch.Metric({
      metricName: 'SelectThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    const dbDeleteThroughputMetric = new cloudwatch.Metric({
      metricName: 'DeleteThroughput',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Sum',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
    return {
      dbInsertThroughputMetric,
      dbUpdateThroughputMetric,
      dbSelectThroughputMetric,
      dbDeleteThroughputMetric,
    };
  }

  metricBufferCacheHitRatio(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'BufferCacheHitRatio',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }

  metricDbConnections(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'DatabaseConnections',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }

  metricReplicaLag(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'AuroraReplicaLag',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }

  metricCpuUtilization(clusterIdentifier: string) {
    return new cloudwatch.Metric({
      metricName: 'CPUUtilization',
      namespace: RdsAuroraMetricFactory.Namespace,
      period: cdk.Duration.minutes(5),
      statistic: 'Average',
      dimensions: {
        DBClusterIdentifier: clusterIdentifier,
      },
    });
  }
}
