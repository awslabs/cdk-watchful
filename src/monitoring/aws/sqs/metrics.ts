import { Metric, Statistic } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';

const enum Metrics {
  ApproximateNumberOfMessagesVisible = 'ApproximateNumberOfMessagesVisible',
  NumberOfMessagesSent = 'NumberOfMessagesSent',
  ApproximateAgeOfOldestMessage = 'ApproximateAgeOfOldestMessage',
  SentMessageSize = 'SentMessageSize'
}

const Namespace = 'AWS/SQS';

export class SqsMetricFactory {
  metricApproximateVisibleMessages(topicName: string) {
    return this.metric(Metrics.ApproximateNumberOfMessagesVisible, topicName).with({ statistic: Statistic.MAXIMUM });
  }

  metricIncomingMessages(topicName: string) {
    return this.metric(Metrics.NumberOfMessagesSent, topicName).with({ statistic: Statistic.SUM });
  }

  metricAgeOfOldestMessageInSeconds(topicName: string) {
    return this.metric(Metrics.ApproximateAgeOfOldestMessage, topicName).with({ statistic: Statistic.MAXIMUM });
  }

  metricAverageMessageSizeInBytes(topicName: string) {
    return this.metric(Metrics.SentMessageSize, topicName).with({ statistic: Statistic.AVERAGE });
  }

  protected metric(metric: Metrics, topicName: string) {
    return new Metric({
      metricName: metric,
      namespace: Namespace,
      period: Duration.minutes(5),
      dimensions: {
        TopicName: topicName,
      },
    });
  }
}
