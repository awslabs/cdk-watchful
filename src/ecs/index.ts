import { Names } from 'aws-cdk-lib';
import * as ecs from 'aws-cdk-lib/aws-ecs';
import { ApplicationTargetGroup } from 'aws-cdk-lib/aws-elasticloadbalancingv2';
import { Construct } from 'constructs';
import { IWatchful } from '../api';
import { linkForEcsService, WatchService } from './service';
import { WatchTargetGroup } from './target-group';

export * from './target-group';
export * from './service';


export interface WatchEcsServiceOptions {
  /**
     * Threshold for the Cpu Maximum utilization
     *
     * @default 80
     */
  readonly cpuMaximumThresholdPercent?: number;

  /**
   * Threshold for the Memory Maximum utilization.
   *
   * @default - 0.
   */
  readonly memoryMaximumThresholdPercent?: number;

  /**
   * Threshold for the Target Response Time.
   *
   * @default - 0.
   */
  readonly targetResponseTimeThreshold?: number;

  /**
   * Threshold for the Number of Requests.
   *
   * @default - 0.
   */
  readonly requestsThreshold?: number;

  /**
   * Threshold for the Number of Request Errors.
   *
   * @default - 0.
   */
  readonly requestsErrorRateThreshold?: number;
}

export interface WatchEcsServiceProps extends WatchEcsServiceOptions {
  readonly title: string;
  readonly watchful: IWatchful;
  readonly fargateService?: ecs.FargateService;
  readonly ec2Service?: ecs.Ec2Service;
  readonly targetGroup: ApplicationTargetGroup;
}

export class WatchEcsService extends Construct {

  private readonly watchful: IWatchful;
  private readonly ecsService: ecs.BaseService;

  constructor(scope: Construct, id: string, props: WatchEcsServiceProps) {
    super(scope, id);

    this.watchful = props.watchful;

    if (!props.ec2Service || !props.fargateService) {
      throw new Error('No service provided to monitor.');
    };

    this.ecsService = props.ec2Service ?? props.fargateService;

    this.watchful.addSection(props.title, {
      links: [
        { title: 'ECS Service', url: linkForEcsService(this.ecsService) },
      ],
    });

    new WatchService(this, Names.uniqueId(this.ecsService), {
      ...props,
      service: this.ecsService,
      skipLinkSection: true,
    });

    new WatchTargetGroup(this, Names.uniqueId(props.targetGroup), {
      ...props,
      skipLinkSection: true,
    });
  }

}
