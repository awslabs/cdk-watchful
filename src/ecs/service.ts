import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';
import * as ecs from 'aws-cdk-lib/aws-ecs';
import { Construct } from 'constructs';
import { IWatchful } from '../api';
import { WatchEcsServiceOptions } from '../ecs';
import { EcsMetricFactory } from '../monitoring/aws/ecs/metrics';


export interface WatchServiceProps extends WatchEcsServiceOptions {
  readonly title: string;
  readonly watchful: IWatchful;
  readonly service: ecs.BaseService;
  /**
   * Whether to add link section at the start of widget
   * @default - false
   */
  readonly skipLinkSection?: boolean;
}

export class WatchService extends Construct {

  private readonly watchful: IWatchful;
  private readonly ecsService: any;
  private readonly serviceName: string;
  private readonly clusterName: string;
  private readonly metrics: EcsMetricFactory;

  constructor(scope: Construct, id: string, props: WatchServiceProps) {
    super(scope, id);

    this.watchful = props.watchful;

    this.ecsService = props.service;
    this.serviceName = props.service.serviceName;
    this.clusterName = props.service.cluster.clusterName;


    this.metrics = new EcsMetricFactory();

    const skipLinkSection = props.skipLinkSection ?? false;

    if (!skipLinkSection) {
      this.watchful.addSection(props.title, {
        links: [
          { title: 'ECS Service', url: linkForEcsService(this.ecsService) },
        ],
      });
    }

    const { cpuUtilizationMetric, cpuUtilizationAlarm } = this.createCpuUtilizationMonitor(props.cpuMaximumThresholdPercent);
    const { memoryUtilizationMetric, memoryUtilizationAlarm } = this.createMemoryUtilizationMonitor(props.memoryMaximumThresholdPercent);


    this.watchful.addWidgets(
      new cloudwatch.GraphWidget({
        title: `CPUUtilization/${cpuUtilizationMetric.period.toMinutes()}min`,
        width: 12,
        left: [cpuUtilizationMetric],
        leftAnnotations: [cpuUtilizationAlarm.toAnnotation()],
      }),
      new cloudwatch.GraphWidget({
        title: `MemoryUtilization/${memoryUtilizationMetric.period.toMinutes()}min`,
        width: 12,
        left: [memoryUtilizationMetric],
        leftAnnotations: [memoryUtilizationAlarm.toAnnotation()],
      }),
    );
  }

  private createCpuUtilizationMonitor(cpuMaximumThresholdPercent = 0) {
    const cpuUtilizationMetric = this.metrics.metricCpuUtilizationAverage(this.clusterName, this.serviceName);
    const cpuUtilizationAlarm = cpuUtilizationMetric.createAlarm(this, 'cpuUtilizationAlarm', {
      alarmDescription: 'cpuUtilizationAlarm',
      threshold: cpuMaximumThresholdPercent,
      comparisonOperator: cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
      evaluationPeriods: 3,
    });
    this.watchful.addAlarm(cpuUtilizationAlarm);
    return { cpuUtilizationMetric, cpuUtilizationAlarm };
  }

  private createMemoryUtilizationMonitor(memoryMaximumThresholdPercent = 0) {
    const memoryUtilizationMetric = this.metrics.metricMemoryUtilizationAverage(this.clusterName, this.serviceName);
    const memoryUtilizationAlarm = memoryUtilizationMetric.createAlarm(this, 'memoryUtilizationAlarm', {
      alarmDescription: 'memoryUtilizationAlarm',
      threshold: memoryMaximumThresholdPercent,
      comparisonOperator: cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
      evaluationPeriods: 3,
    });
    this.watchful.addAlarm(memoryUtilizationAlarm);
    return { memoryUtilizationMetric, memoryUtilizationAlarm };
  }

}

export function linkForEcsService(ecsService: any) {
  return `https://console.aws.amazon.com/ecs/home?region=${ecsService.stack.region}#/clusters/${ecsService.cluster.clusterName}/services/${ecsService.serviceName}/details`;
}
