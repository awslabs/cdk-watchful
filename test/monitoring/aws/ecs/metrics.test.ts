import { EcsMetricFactory } from '../../../../src/monitoring/aws/ecs/metrics';

const DummyTargetGroup = 'DummyTargetGroup';
const DummyLoadBalancer = 'DummyLoadBalancer';
const DummyClusterName = 'DummyClusterName';
const DummyServiceName = 'DummyServiceName';

test('snapshot test: metricRequestCount', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricRequestCount(DummyTargetGroup, DummyLoadBalancer);

  // THEN
  expect(metric.metricName).toStrictEqual('RequestCount');
  expect(metric.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricTargetResponseTime', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricTargetResponseTime(DummyTargetGroup, DummyLoadBalancer);

  // THEN
  Object.values(metric).forEach(eachMetric => {
    expect(eachMetric.metricName).toStrictEqual('TargetResponseTime');
    expect(eachMetric.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  });
  expect(metric).toMatchSnapshot();
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricCpuUtilizationAverage', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricCpuUtilizationAverage(DummyClusterName, DummyServiceName);

  // THEN
  expect(metric.metricName).toStrictEqual('CPUUtilization');
  expect(metric.dimensions).toStrictEqual({ ClusterName: DummyClusterName, ServiceName: DummyServiceName });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricMemoryUtilizationAverage', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricMemoryUtilizationAverage(DummyClusterName, DummyServiceName);

  // THEN
  expect(metric.metricName).toStrictEqual('MemoryUtilization');
  expect(metric.dimensions).toStrictEqual({ ClusterName: DummyClusterName, ServiceName: DummyServiceName });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricMinHealthyHostCount', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricMinHealthyHostCount(DummyTargetGroup, DummyLoadBalancer);

  // THEN
  expect(metric.metricName).toStrictEqual('HealthyHostCount');
  expect(metric.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricMaxUnhealthyHostCount', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricMaxUnhealthyHostCount(DummyTargetGroup, DummyLoadBalancer);

  // THEN
  expect(metric.metricName).toStrictEqual('UnHealthyHostCount');
  expect(metric.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricHttpErrorStatusCodeRate', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricHttpErrorStatusCodeRate(DummyTargetGroup, DummyLoadBalancer);

  // THEN
  expect(metric).toMatchSnapshot();
});

test('snapshot test: metricHttpStatusCodeCount', () => {
  // GIVEN
  const unitToTest = new EcsMetricFactory();

  // WHEN
  const metric = unitToTest.metricHttpStatusCodeCount(DummyTargetGroup, DummyLoadBalancer);

  // THEN
  expect(metric.count2XX.metricName).toStrictEqual('HTTPCode_Target_2XX_Count');
  expect(metric.count2XX.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric.count3XX.metricName).toStrictEqual('HTTPCode_Target_3XX_Count');
  expect(metric.count3XX.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric.count4XX.metricName).toStrictEqual('HTTPCode_Target_4XX_Count');
  expect(metric.count4XX.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric.count5XX.metricName).toStrictEqual('HTTPCode_Target_5XX_Count');
  expect(metric.count5XX.dimensions).toStrictEqual({ TargetGroup: DummyTargetGroup, LoadBalancer: DummyLoadBalancer });
  expect(metric).toMatchSnapshot();
});
