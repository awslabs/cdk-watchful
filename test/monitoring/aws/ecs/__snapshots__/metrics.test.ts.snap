// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: metricCpuUtilizationAverage 1`] = `
Metric {
  "account": undefined,
  "color": undefined,
  "dimensions": Object {
    "ClusterName": "DummyClusterName",
    "ServiceName": "DummyServiceName",
  },
  "label": undefined,
  "metricName": "CPUUtilization",
  "namespace": "AWS/ECS",
  "period": Duration {
    "amount": 5,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "region": undefined,
  "statistic": "Average",
  "unit": undefined,
}
`;

exports[`snapshot test: metricHttpErrorStatusCodeRate 1`] = `
MathExpression {
  "color": undefined,
  "expression": "http4xx + http5xx / requests",
  "label": undefined,
  "period": Duration {
    "amount": 5,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "usingMetrics": Object {
    "http4xx": Metric {
      "account": undefined,
      "color": undefined,
      "dimensions": Object {
        "LoadBalancer": "DummyLoadBalancer",
        "TargetGroup": "DummyTargetGroup",
      },
      "label": undefined,
      "metricName": "HTTPCode_Target_4XX_Count",
      "namespace": "AWS/ApplicationELB",
      "period": Duration {
        "amount": 5,
        "unit": TimeUnit {
          "inMillis": 60000,
          "isoLabel": "M",
          "label": "minutes",
        },
      },
      "region": undefined,
      "statistic": "Sum",
      "unit": undefined,
    },
    "http5xx": Metric {
      "account": undefined,
      "color": undefined,
      "dimensions": Object {
        "LoadBalancer": "DummyLoadBalancer",
        "TargetGroup": "DummyTargetGroup",
      },
      "label": undefined,
      "metricName": "HTTPCode_Target_5XX_Count",
      "namespace": "AWS/ApplicationELB",
      "period": Duration {
        "amount": 5,
        "unit": TimeUnit {
          "inMillis": 60000,
          "isoLabel": "M",
          "label": "minutes",
        },
      },
      "region": undefined,
      "statistic": "Sum",
      "unit": undefined,
    },
    "requests": Metric {
      "account": undefined,
      "color": undefined,
      "dimensions": Object {
        "LoadBalancer": "DummyLoadBalancer",
        "TargetGroup": "DummyTargetGroup",
      },
      "label": undefined,
      "metricName": "RequestCount",
      "namespace": "AWS/ApplicationELB",
      "period": Duration {
        "amount": 5,
        "unit": TimeUnit {
          "inMillis": 60000,
          "isoLabel": "M",
          "label": "minutes",
        },
      },
      "region": undefined,
      "statistic": "Sum",
      "unit": undefined,
    },
  },
}
`;

exports[`snapshot test: metricHttpStatusCodeCount 1`] = `
Object {
  "count2XX": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "HTTPCode_Target_2XX_Count",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
  "count3XX": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "HTTPCode_Target_3XX_Count",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
  "count4XX": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "HTTPCode_Target_4XX_Count",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
  "count5XX": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "HTTPCode_Target_5XX_Count",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Sum",
    "unit": undefined,
  },
}
`;

exports[`snapshot test: metricMaxUnhealthyHostCount 1`] = `
Metric {
  "account": undefined,
  "color": undefined,
  "dimensions": Object {
    "LoadBalancer": "DummyLoadBalancer",
    "TargetGroup": "DummyTargetGroup",
  },
  "label": undefined,
  "metricName": "UnHealthyHostCount",
  "namespace": "AWS/ApplicationELB",
  "period": Duration {
    "amount": 5,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "region": undefined,
  "statistic": "Maximum",
  "unit": undefined,
}
`;

exports[`snapshot test: metricMemoryUtilizationAverage 1`] = `
Metric {
  "account": undefined,
  "color": undefined,
  "dimensions": Object {
    "ClusterName": "DummyClusterName",
    "ServiceName": "DummyServiceName",
  },
  "label": undefined,
  "metricName": "MemoryUtilization",
  "namespace": "AWS/ECS",
  "period": Duration {
    "amount": 5,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "region": undefined,
  "statistic": "Average",
  "unit": undefined,
}
`;

exports[`snapshot test: metricMinHealthyHostCount 1`] = `
Metric {
  "account": undefined,
  "color": undefined,
  "dimensions": Object {
    "LoadBalancer": "DummyLoadBalancer",
    "TargetGroup": "DummyTargetGroup",
  },
  "label": undefined,
  "metricName": "HealthyHostCount",
  "namespace": "AWS/ApplicationELB",
  "period": Duration {
    "amount": 5,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "region": undefined,
  "statistic": "Minimum",
  "unit": undefined,
}
`;

exports[`snapshot test: metricRequestCount 1`] = `
Metric {
  "account": undefined,
  "color": undefined,
  "dimensions": Object {
    "LoadBalancer": "DummyLoadBalancer",
    "TargetGroup": "DummyTargetGroup",
  },
  "label": undefined,
  "metricName": "RequestCount",
  "namespace": "AWS/ApplicationELB",
  "period": Duration {
    "amount": 5,
    "unit": TimeUnit {
      "inMillis": 60000,
      "isoLabel": "M",
      "label": "minutes",
    },
  },
  "region": undefined,
  "statistic": "Sum",
  "unit": undefined,
}
`;

exports[`snapshot test: metricTargetResponseTime 1`] = `
Object {
  "avg": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "TargetResponseTime",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Average",
    "unit": undefined,
  },
  "max": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "TargetResponseTime",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Maximum",
    "unit": undefined,
  },
  "min": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "TargetResponseTime",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Minimum",
    "unit": undefined,
  },
}
`;

exports[`snapshot test: metricTargetResponseTime 2`] = `
Object {
  "avg": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "TargetResponseTime",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Average",
    "unit": undefined,
  },
  "max": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "TargetResponseTime",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Maximum",
    "unit": undefined,
  },
  "min": Metric {
    "account": undefined,
    "color": undefined,
    "dimensions": Object {
      "LoadBalancer": "DummyLoadBalancer",
      "TargetGroup": "DummyTargetGroup",
    },
    "label": undefined,
    "metricName": "TargetResponseTime",
    "namespace": "AWS/ApplicationELB",
    "period": Duration {
      "amount": 5,
      "unit": TimeUnit {
        "inMillis": 60000,
        "isoLabel": "M",
        "label": "minutes",
      },
    },
    "region": undefined,
    "statistic": "Minimum",
    "unit": undefined,
  },
}
`;
